### PR TITLE
Testing with GHA

### PR DESCRIPTION
**Pull request by ekingery**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: Run linter and tests
+
+on:
+  push:
+    branches:
+      - '*'
```
Comment (ErikSchierboom) [inline .github/workflows/test.yml:6]:
```suggestion
      - main
      - master
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  pull_request:
+    branches:
+      - '*'
```
Comment (ErikSchierboom) [inline .github/workflows/test.yml:9]: You don't need to specify any branches.

```suggestion
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
+name: Run linter and tests
+
+on:
+  push:
```
Comment (ErikSchierboom) [inline .github/workflows/test.yml:4]: This allows the workflow to be triggered manually.

```suggestion
  workflow_dispatch:
  push:
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (ErikSchierboom) [reply .github/workflows/test.yml:6]: This prevents running the workflow twice when a push is made to a pull request's branch.

**Review by ErikSchierboom — APPROVED** [review]

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -1,12 +1,12 @@
 name: Run linter and tests
 
 on:
+  workflow_dispatch:
   push:
     branches:
-      - '*'
+      - main
+      - master
```
Comment (ErikSchierboom) [inline .github/workflows/test.yml:8]: I've just had some feedback that we should not restrict the branches. Sorry about. That.

You could do:

`on: [push, pull_request, workflow_dispatch]` 

Whatever you prefer

Comment (ekingery) [reply .github/workflows/test.yml:8]: Oops, just merged this. I'll fix in another #23

**Pull request merged**